dotCMS/src/main/java/com/dotcms/rest/api/v1/template/TemplateResource.java

```diff
@@ -0,0 +1,752 @@
+        }
+
+        this.templateHelper.checkPermission(user, template, PERMISSION_WRITE);
+        final boolean result = this.templateAPI.archive(template, user, pageMode.respectAnonPerms);
+
+        Logger.debug(this, "Archive done template: " + templateInode);
+        return Response.ok(new ResponseEntityView(result)).build();
+    }
+
+    /**
+     * Unarchive a template
+     * @param request            {@link HttpServletRequest}
+     * @param response           {@link HttpServletResponse}
+     * @param templateInode      {@link String} template inode to unlock
+     * @return Response
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @PUT
+    @Path("/{templateInode}/_unarchive")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response unarchive(@Context final HttpServletRequest  request,
+                                  @Context final HttpServletResponse response,
+                                  @PathParam("templateInode") final String templateInode) throws DotDataException, DotSecurityException {
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(request, response).rejectWhenNoUser(true).init();
+        final User user         = initData.getUser();
+        final PageMode pageMode = PageMode.get(request);
+
+        Logger.debug(this, ()->"Doing unarchive of the Template: " + templateInode);
+
+        final Template template = this.templateAPI.find(templateInode, user, pageMode.respectAnonPerms);
+
+        if (null == template || !InodeUtils.isSet(template.getInode())) {
+
+            throw new DoesNotExistException("The  template inode: " + templateInode + " does not exists");
+        }
+
+        this.templateHelper.checkPermission(user, template, PERMISSION_WRITE);
+        this.templateAPI.unarchive(template);
+
+        Logger.debug(this, "Unarchive done template: " + templateInode);
+        return Response.ok(new ResponseEntityView(true)).build();
+    }
+
+    /**
+     * Deletes a template
+     * Pre: template must not has dependencies
+     * @param request            {@link HttpServletRequest}
+     * @param response           {@link HttpServletResponse}
+     * @param templateInode      {@link String} template inode to look for the template and then, delete it
+     * @return Response
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @DELETE
+    @Path("/{templateInode}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response delete(@Context final HttpServletRequest  request,
+                                    @Context final HttpServletResponse response,
+                                    @PathParam("templateInode") final String templateInode) throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)